### PR TITLE
ROX-19821: Adding exception management page to UI

### PR DESCRIPTION
**Pull request by sachaudh**
## Description

This PR adds a route and page for `Exception Management`

Screenshot 2023-10-19 at 12 57 16 PM

### CODE REVIEW

**Comment by sachaudh** [comment]
According to the mocks, there is a separation between the Workload CVEs nav item and the Exception Management and Reporting nav items

Screenshot 2023-10-19 at 1 00 31 PM

The current implementation of the navigation sidebar doesn't allow for this divider so we will have to modify the sidebar logic to allow a divider. For now, I'll just add the nav item without a divider. I'll send out a draft PR for that change and include Mark as a reviewer (since I believe he was involved in refactoring the navigation sidebar in the past)

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx
Comment (dvail) [inline ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx:12]: I believe we need a fourth tab for "Denied" as well https://docs.google.com/presentation/d/1PCYmIS9SkRr9lnZdPJ4bdEkFD3cdq2vYZExq0P_ltn4/edit?pli=1#slide=id.g27eff3a2f70_0_12.

From what I can tell in the requirements doc we need to include this page, but the ability to re-open denied requests will be post-MVP.

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx
Comment (dvail) [inline ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx:15]: What do you think about using the `useURLStringUnion` hook instead, so that we can track the tab in the URL and allow linking to the different sections?

File: ui/apps/platform/src/routePaths.ts
Comment (dvail) [inline ui/apps/platform/src/routePaths.ts:254]: I think the minimum required resource for this page should be `VulnerabilityManagementRequests`. A user who opens a deferral/FP should be able to view this page, and they might not have access to the `Administration` resource. We might even need to distinguish between READ_ACCESS and READ_WRITE_ACCESS for regular requester level users inside this page.

Once we get into the page itself, approval based features will also require conditional rendering depending on the user's access to `VulnerabilityManagementApprovals`.

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx
Comment (dvail) [inline ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx:22]: Small nit: can we add a `<PageTitle>` component here as well?

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx
Comment (dvail) [inline ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx:28]: Small small nit:
```suggestion
                    <Title headingLevel="h1">Exception management</Title>
```

**Review by dvail — APPROVED** [review]
LGTM - a few very small things that I think we should change but up to you if you want to do it in this PR or a follow up.

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx
Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx:12]: Good point! I didn't even notice the updated slides section. Was wondering why the slides looked unchanged. Will add that as well

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx
Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx:22]: I decided not to add it here because we'll be adding different sections (ie. Pending approvals, etc.) so I was thinking of adding it to those sections

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx
Comment (dvail) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx:12]: Yeah unfortunately I think the linked slide is the _only_ slide with the change. 🤷‍♂️

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionManagementPage.tsx
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionManagementPage.tsx:11]: What do y’all think about moving toward a convention for container sub-routing component?

Page seems ambiguous with the requests page and a request page.

As a starting point for discussion, and I expect can be improved on:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Administration/Events/AdministrationEventsRoute.tsx

By the way, because this example did not use `Switch` element and because React Router 6 has `Routes` element instead, I did not use **Switch**. However, I am not sure how well **Route** reads.

Here is another place to consider for intuitive naming:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/MainPage/Body.tsx#L73-L78

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx:12]: This answers a question in my mind: Deferral Management includes both deferrals and false positives.

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx:28]: Should we discuss as team and maybe also double check at an upcoming PatternFly meeting where is the boundary between Title Case in sidebar and Sentence case for the same thing as page title and heading.

File: ui/apps/platform/src/routePaths.ts
Comment (pedrottimark) [inline ui/apps/platform/src/routePaths.ts:45]: * Plus: path segment is exception-management consistent with recently added exception-configuration
* Minus: title **Deferral Management** is inconsistent with both its own route and its configuration. Would **Exception Management** be more consistent and also remove ambiguity of **Deferral** referring to both deferral and false positive requests?

File: ui/apps/platform/src/routePaths.ts
Comment (pedrottimark) [inline ui/apps/platform/src/routePaths.ts:417]: Ditto comment by Dave to render `PageTitle` element, which makes this unneeded, although consistent.

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx
Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionRequestsPage.tsx:12]: @pedrottimark I think they've decided to rename it to Exception Management now

File: ui/apps/platform/src/routePaths.ts
Comment (sachaudh) [reply ui/apps/platform/src/routePaths.ts:254]: If a user has access to `VulnerabilityManagementApprovals`, that should imply they have access to `VulnerabilityManagementRequests` as well, right? Does it automatically give access to `VulnerabilityManagementRequests` though or will we need to do an `OR` for both of those permissions?

File: ui/apps/platform/src/routePaths.ts
Comment (sachaudh) [reply ui/apps/platform/src/routePaths.ts:45]: Yes, I think we have decided to go with Exception Management so it makes more sense in terms of including both deferral and false positive requests

File: ui/apps/platform/src/routePaths.ts
Comment (dvail) [reply ui/apps/platform/src/routePaths.ts]: Hmm that I'm not sure. We would have to check with the BE team to verify. My gut feeling is you are right, that a user should have either `OR` permission to access the page.

It certainly seems possible that someone would configure a user with `VulnerabilityManagementApprovals`, but not `VulnerabilityManagementRequests`, so it makes sense to check either one for the top level routing. We will need to confirm what the API behavior is though to handle how the UI is displayed inside the page.

File: ui/apps/platform/src/Containers/MainPage/Sidebar/NavigationSidebar.tsx
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/MainPage/Sidebar/NavigationSidebar.tsx:35]: Not a blocker to merge, because I can move in follow up for separator, according to comment above:

```js
// Import path variables in alphabetical order to minimize merge conflicts when multiple people add routes.
```

File: ui/apps/platform/src/routePaths.ts
Comment (sachaudh) [reply ui/apps/platform/src/routePaths.ts]: I'll add both (with an OR) for now and maybe ask about it in our VM sync tomorrow.

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionManagementPage.tsx
Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionManagementPage.tsx:11]: @pedrottimark I noticed that `AdministrationEventsRoute` isn't referenced anywhere. Is that a bug?

Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionManagementPage.tsx:11]: Good eyew. Because dynamic loading is only reference:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/MainPage/Body.tsx#L73-L78

Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionManagementPage.tsx:11]: Ahh, I see. That actually raises a good point about cleaning up code. What if someone notices that something isn't referenced and decides to delete that bit of code? That might cause some issues unless the code is tested well. I guess that gives more of a reason to test dynamically generated content.

Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionManagementPage.tsx:11]: Also, on the subject of `Page` vs. `Route`. Was there a discussion about this before? It seems we have a few that use `Route`, but a majority use `Page`. What's your reasoning for using the term `Route`, @pedrottimark? Personally, I'm more inclined to go with `Page` because it describes the component better. In terms of sub-routing, I think I have a PR coming out that does make use of sub-routing. Maybe we can discuss if that approach works or not in that PR. For now, I'll go ahead and merge this, but let's keep the discussion going

Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/ExceptionManagementPage.tsx:11]: Yeah, no worries. For administration events, there are 2 page elements, so I ran out of names :)
* AdministrationEventsPage
* AdministrationEventPage

File: ui/apps/platform/src/Containers/MainPage/Sidebar/NavigationSidebar.tsx
Comment (sachaudh) [reply ui/apps/platform/src/Containers/MainPage/Sidebar/NavigationSidebar.tsx:35]: @pedrottimark let's move it in your PR

**Pull request merged**
